### PR TITLE
cspell lint go code

### PR DESCRIPTION
**Pull request by 6543**
address  #3705

### CODE REVIEW

**Comment by xoxys** [comment]
In general, I like that we add spellchecking to more areas. But is there no way to handle excludes in a central place? E.g. most imports could be excluded by a `ignoreRegExpList` setting?

**Comment by 6543** [comment]
Well it is handled in one place at .cspell.json , or what do you intend?

**Comment by xoxys** [comment]
I'm talking about the `// cspell:` annotation behind a lot of import URLs.

**Comment by 6543** [comment]
well yes because the usernames are mostly not in a dictionary and I don't want to add them to ours as we don't need them globaly

**Comment by xoxys** [comment]
That's why I asked for a central `ignoreRegExpList` config and not dict entry. Basically something like:

```
"ignoreRegExpList": [
    ".*github.com/.*"
]
```

URLs should not be linted at all.

**Comment by xoxys** [comment]
Here is an example config https://gist.github.com/r3code/21d1e9a3f862ad865808f07225b59068

File: .woodpecker/social.yaml [hunk 1]
```diff
@@ -25,7 +25,7 @@ steps:
 
         The changelog entry: {{ changelog }}
   - name: discord
-    image: docker.io/appleboy/drone-discord:1.3.2
+    image: docker.io/appleboy/drone-discord:1.3.2 # cspell:disable-line
```
Comment (xoxys) [inline .woodpecker/social.yaml:28]: Can we add a regex for container image urls aswell?

File: .woodpecker/test.yaml [hunk 2]
```diff
@@ -74,7 +74,7 @@ steps:
   lint-license-header:
     image: *golang_image
     commands:
-      - go install github.com/google/addlicense@latest
+      - go install github.com/google/addlicense@latest # cspell:words addlicense
```
Comment (xoxys) [inline .woodpecker/test.yaml:77]: Same here, exlude all go install commands by a regex

File: agent/runner.go [hunk 3]
```diff
@@ -50,11 +50,11 @@ func NewRunner(workEngine rpc.Peer, f rpc.Filter, h string, state *State, backen
 	}
 }
 
-func (r *Runner) Run(runnerCtx context.Context) error { //nolint:contextcheck
+func (r *Runner) Run(runnerCtx context.Context) error { //nolint:contextcheck // cspell:words contextcheck
```
Comment (xoxys) [inline agent/runner.go:53]: All nolint annotations should be exckuded by a regex

File: pipeline/backend/kubernetes/utils.go [hunk 4]
```diff
@@ -65,23 +65,23 @@ func isImagePullBackOffState(pod *v1.Pod) bool {
 	return false
 }
 
-// getClientOutOfCluster returns a k8s clientset to the request from outside of cluster.
+// getClientOutOfCluster returns a k8s client set to the request from outside of cluster.
 func getClientOutOfCluster() (kubernetes.Interface, error) {
-	kubeConfigPath := os.Getenv("KUBECONFIG")
+	kubeConfigPath := os.Getenv("KUBECONFIG") // cspell:words KUBECONFIG
```
Comment (xoxys) [inline pipeline/backend/kubernetes/utils.go:70]: Common, could be added to dictionary.

File: pipeline/backend/kubernetes/utils.go [hunk 5]
```diff
@@ -23,7 +23,7 @@ import (
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/clientcmd"
+	client_cmd "k8s.io/client-go/tools/clientcmd" // cspell:disable-line
```
Comment (xoxys) [inline pipeline/backend/kubernetes/utils.go:26]: Why does this line need to be excluded? Shouldnt it be covered by the regex exclude for import statements?

File: pipeline/backend/local/command.go [hunk 6]
```diff
@@ -50,11 +50,12 @@ func (e *local) genCmdByShell(shell string, cmds []string) (args []string, err e
 		return []string{"/c", cmd.Name()}, nil
 	case "fish":
 		script := ""
-		for _, cmd := range cmds {
+		for _, cmd := range cmdList {
 			script += fmt.Sprintf("echo %s\n%s || exit $status\n", strings.TrimSpace(shellescape.Quote("+ "+cmd)), cmd)
 		}
 		return []string{"-c", script}, nil
-	case "powershell", "pwsh":
+	case "powershell", "pwsh": // cspell:words pwsh
```
Comment (xoxys) [inline pipeline/backend/local/command.go:57]: pwsh is afaik also common and could be added to the dict.

File: .woodpecker/test.yaml (hunk 2 above)
Comment (6543) [reply .woodpecker/test.yaml:77]: yes but the next lines not

File: pipeline/backend/kubernetes/utils.go (hunk 4 above)
Comment (6543) [reply pipeline/backend/kubernetes/utils.go:70]: image

not realy common i would argue

**Review by xoxys — APPROVED** [review]

**Pull request merged**